numpy/core/tests/test_dtype.py

```diff
@@ -883,3 +886,17 @@ class PaddedStruct(ctypes.BigEndianStructure):
             ('b', '>H')
         ], align=True)
         self.check(PaddedStruct, expected)
+
+    def test_big_endian_simple_type(self):
+        byte_order = np.dtype(ctypes.c_uint16.__ctype_be__).byteorder
+        if sys.byteorder == 'little':
+            expected = '>'
+        else:
+            expected = '='
+        assert byte_order == expected
+
+    def test_simple_endian_types(self):
+        assert_equal(np.dtype(ctypes.c_uint16.__ctype_le__), np.dtype('<u2'))
+        assert_equal(np.dtype(ctypes.c_uint16.__ctype_be__), np.dtype('>u2'))
+        assert_equal(np.dtype(ctypes.c_uint8.__ctype_le__), np.dtype('u1'))
+        assert_equal(np.dtype(ctypes.c_uint8.__ctype_be__), np.dtype('u1'))
```

Comment:
You should actually use `self.check` here, it tests both types and instances.

Reply:
Done.

Reply:
Ok, I just had to look how `check `was defined... Did the commit now.